### DISCUSSION TITLE
dotMemoryUnit & cake-contrib

### DISCUSSION THREAD

**Discussion by mgnslndh** [cake-build/cake#4235] [Q&A]
I'm considering open sourcing my Cake tool for JetBrain's [dotMemoryUnit ](https://www.jetbrains.com/dotmemory/unit/). 

Is this a good fit for the cake-contrib organization or should I host it in my own repository? What are requirement or process for submitting repositories to be part of the cake-contrib organization?

**Comment by pascalberger** [comment]
Yes, that's definitely a fit.

The process is documented at https://cakebuild.net/community/cake-contrib and https://github.com/cake-contrib/Home/blob/master/CONTRIBUTING.md for details.

Do you prefer to implement the addin first in our own repo and move it to cake-contrib when ready, or have the repo available in cake-contrib org from the beginning?

**Reply by mgnslndh** [reply]
Great! Thanks for the links. I think that the first step would be to migrate the existing code to a new public repo under my user. When that is done we could move it to cake-contrib. At that point I will probably need some help or guidance on all the CI/publishing stuff required. Never done that on GitHub before and I guess there are some conventions to follow for cake-contrib.

**Reply by pascalberger** [reply]
Perfect 👍 Let us know here when you're ready to move the repo the `cake-contrib` organisation and we can walk with you through the process.

**Comment by mgnslndh** [comment]
I've made the migration from my private repo to a public repo https://github.com/mgnslndh/Cake.DotMemoryUnit. It is not much original work since I based the tool on the DotCover tool from Cake.Common. They are very similar in the way they need to intercept & control another tool execution. At some point the tool might be ready for the Cake repo instead.

So, what is left to do? 

- Building and publishing according to Cake conventions
- Fix missing XML docs and generated documentation
- Add support for additional command line options
  - Only 2/11 are supported since that was all I needed for my scenario

Should I add more command line options and fix XML docs now or are we ready to move to another repo?

**Reply by mgnslndh** [reply]
@pascalberger do you have any feedback on this? Should I focus on adding more command line options or will a minimal feature set work for the initial version?

**Reply by pascalberger** [reply]
@mgnslndh Sorry for the late reply. It is perfectly fine to transfer to `cake-contrib` org like this and finish it afterwards there.

Here are the instructions for the transfer:

For GitHub, this is a multi-step operation due to how GitHub manages permissions for organizations.

1. Verify that I'm a member of the core Cake maintainer team on the official website: https://cakebuild.net/docs/team/
2. Transfer the repo to my GitHub user `pascalberger` ([instructions](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository))
3. I transfer it to the [Cake Contrib](https://github.com/cake-contrib) organization
4. You accept the invite to join the organization (you should have received an email)
5. I add you as admin in your repo, within the org

Once the addin is pubslished to NuGet, open the "Manage" package UI for the Cake.DotMemoryUnit package on [nuget.org](http://nuget.org/) and add the user `cake-contrib` as owner

**Reply by mgnslndh** [reply]
@pascalberger No worries, there is no rush :) I've made the transfer.

**Reply by pascalberger** [reply]
@mgnslndh The repository has been transferred to the cake-contrib organization and you should have received invitations to join the org and team.

Do you plan to use AppVeyor?

**Reply by mgnslndh** [reply]
@pascalberger I have no prior experience of AppVeyor but I would like for the repo to have a CI build so if that is how Cake/Cake-Contrib does CI I would like to use it.

**Reply by pascalberger** [reply]
@mgnslndh You're free to use whatever you prefer. Historically a lot of repos used AppVeyor, but we also have an Azure Pipelines tenant or some projects are using GitHub Actions (e.g. https://github.com/cake-contrib/Cake.ESLint) or a combination of all of them (e.g. https://github.com/cake-contrib/Cake.Issues.Recipe)

**Reply by mgnslndh** [reply]
@pascalberger I settled on GitHub Actions for CI. However, I'm having some trouble when the build runs the recipe target "Publish-PreRelease-Packages". It tries to upload a preview package to a NuGet source on Azure but accessing that source is denied because of an authorization issue. The recipe/action is based on Cake.ESLint. Do you have any pointers on how to proceed? 

https://github.com/cake-contrib/Cake.DotMemoryUnit/actions/runs/11304629640/job/31443989411

**Reply by nils-a** [reply]
@mgnslndh it seems I'm having the same problem in Cake.ESLint, currently:

https://github.com/cake-contrib/Cake.ESLint/actions/runs/11354870042/job/31583028599#step:6:1179

I'm guessing the PAT used to access the feed has expired. 
I'll see if I can find some time tonight to update it.

**Reply by nils-a** [reply]
@mgnslndh can you try re-running that failed build?

**Reply by mgnslndh** [reply]
@nils-a it looks like you solved the problem. The package is now pushed but rejected because it is a duplicate. I need to look into versioning. Thanks!
